Updated redux-persist's transform functions

kshavp opened the pull request:
Removed the old field 'phishing' and fixed the typos.

<!--
Thanks for your contribution!

Please ensure that any applicable requirements below are satisfied before submitting this pull request. This will help ensure a quick and efficient review cycle.
-->

**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/coding_guidelines/CODING_GUIDELINES.md)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add the appropiate QA label when dev review is completed
    - `needs-qa`: PR requires manual QA.
    - `No QA/E2E only`: PR does not require any manual QA effort. Prior to merging, ensure that you have successful end-to-end test runs in Bitrise. 
    - `Spot check on release build`: PR does not require feature QA but needs non-automated verification. In the description section, provide test scenarios. Add screenshots, and or recordings of what was tested.
5. Add `QA Passed` label when QA has signed off (Only required if the PR was labeled with `needs-qa`)
6. Add your team's label, i.e. label starting with `team-` (or `external-contributor` label if your not a MetaMask employee)

**Description**

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_
_1. What is the reason for the change?_
_2. What is the improvement/solution?_

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

fixes #7204

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

kshavp committed:
Updated redux-persist's transform functions

Removed the old field 'phishing' and fixed the typos.

Gudahtt commented on app/store/index.ts:
```diff
@@ -90,7 +90,7 @@ const persistTransform = createTransform(
     } = SwapsController;
     // TODO: Fix this type error
     // @ts-expect-error There is no `phishing` property in the phishing controller state
```
Each of the two changed lines has a TODO comment and a ts-expect-error comment. Could you remove all of these? They are no longer relevant now that you have fixed the typo and removed that field. The `ts-expect-error` line will cause a type error as well.

kshavp replied:
Removed! Thanks.

kshavp committed:
Todo Comments removed

Removed the TODO comment and a ts-expect-error comment which is no longer required.

kshavp committed:
Merge branch 'main' into kshavp-transformFunctionCleanup
